wordpress-mobile/WordPress-Android
# Add analytics event for tracking Jetpack full screen overlay events

Part of #17333 

## Changes

This PR adds the logic for tracking the events related to the FullScreen overlay shown on Stats/Reader and Notifications. 


## Review instructions 
Approval from one reviewer is sufficient for merging this PR. 

## To test:

### Pre- Requisites
- Launch WP app -> Go to my site -> App settings -> Debug settings -> Enable jp_removal_one flag
- Navigate to Me > App Settings > Privacy Setting > and enable Collect Information

Test 1 - Stats - Close overlay with Continue to feature button 
-  Launch the app -> Go to my site -> Click on stats
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"stats"}
- Click at **Continue to stats** button 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_dismissed, Properties: {"phase":"one","source":"stats","dismissal_type":"continue"}


Test 2 - Stats - Close overlay with X button 
-  Launch the app -> Go to my site -> Click on stats
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"stats"}
- Click at the Close(x) button on the top left corner 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_dismissed, Properties: {"phase":"one","source":"stats","dismissal_type":"close"}


Test 3 - Stats - Try jetpack app flow 
-  Launch the app -> Go to my site -> Click on stats
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"stats"}
- Click at **Try the new jetpack app** button 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_button_tapped, Properties: {"phase":"one","source":"stats"}

Test 4 - Notifications - Close overlay with Continue to feature button 
-  Launch the app -> Go to my site -> Click on Notifications
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"notifications"}
- Click at **Continue to Notifications** button 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_dismissed, Properties: {"phase":"one","source":"notifications","dismissal_type":"continue"}


Test 5 - Notifications - Close overlay with X button 
-  Launch the app -> Go to my site -> Click on Notifications
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"notifications"}
- Click at the Close(x) button on the top left corner 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_dismissed, Properties: {"phase":"one","source":"notifications","dismissal_type":"close"}


Test 6 - Notifications - Try jetpack app flow 
-  Launch the app -> Go to my site -> Click on Notifications
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"notifications"}
- Click at **Try the new jetpack app** button 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_button_tapped, Properties: {"phase":"one","source":"notifications"}

Test 7 - Reader - Close overlay with Continue to feature button 
-  Launch the app -> Go to my site -> Click on Reader tab
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"reader"}
- Click at **Continue to Reader** button 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_dismissed, Properties: {"phase":"one","source":"reader","dismissal_type":"continue"}


Test 8 - Reader - Close overlay with X button 
-  Launch the app -> Go to my site -> Click on Reader tab
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"reader"}
- Click at the Close(x) button on the top left corner 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_dismissed, Properties: {"phase":"one","source":"reader","dismissal_type":"close"}


Test 9 - Reader - Try jetpack app flow 
-  Launch the app -> Go to my site -> Click on Reader tab
-  ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_displayed, Properties: {"phase":"one","source":"reader"}
- Click at **Try the new jetpack app** button 
- ✅  Verify that the log contains  🔵 Tracked: remove_feature_overlay_button_tapped, Properties: {"phase":"one","source":"reader"}


## ⚠️ Merge Instructions
- [x] Wait for https://github.com/wordpress-mobile/WordPress-Android/pull/17510 to be merged
- [ ] Revert the commit added for testing purposes [6d092ae](https://github.com/wordpress-mobile/WordPress-Android/commit/6d092ae2fced9abf5e1cb6467aa3637d52d8f234)
- [ ] Remove not read for merge label
- [ ]  Merge as normal


## Regression Notes
1. Potential unintended areas of impact
Analytics events are not tracked properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing 

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
